06_puppy/alfredxiao/store_test.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+}
+
+func (s *storerSuite) TestCreatePuppyHappyCase() {
+	t := s.T()
+	err := s.store.CreatePuppy(Puppy{ID: "1", Colour: "Black"})
+	assert.NoError(s.T(), err, "Happy case puppy creation")
+
+	p, _ := s.store.ReadPuppy("1")
+	assert.Equal(t, "Black", p.Colour)
+}
+
+func (s *storerSuite) TestCreatePuppyAlreadyExists() {
+	_ = s.store.CreatePuppy(Puppy{ID: "2"})
+	err := s.store.CreatePuppy(Puppy{ID: "2", Colour: "Red"})
+	assert.Error(s.T(), err, "Puppy creation fails if ID already exists")
+}
+
+func (s *storerSuite) TestReadPuppyHappyCase() {
+	t := s.T()
+	_ = s.store.CreatePuppy(Puppy{ID: "3", Colour: "Blue"})
+	p, err := s.store.ReadPuppy("3")
+	require.NoError(t, err)
```

Comment:
Why the mix of require and assert?

Reply:
require is for not continuing with other assertions when it fails, assertions do not have this attribute. In this case, my thought was that if `err` is `Error` then there is no point in continuing.